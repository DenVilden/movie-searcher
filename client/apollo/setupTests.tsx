import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from '../pages/_app';

// eslint-disable-next-line jest/prefer-spy-on
HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) =>
  render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  );

export * from '@testing-library/react';
