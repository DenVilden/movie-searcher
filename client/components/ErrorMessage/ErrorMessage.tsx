import { Alert, AlertTitle } from '@material-ui/lab';
import styled from 'styled-components';

const StyledAlert = styled(Alert)`
  margin: ${(props) => props.theme.spacing(2, 2, 0, 2)};
`;

interface Props {
  error: string;
}

const ErrorMessage = ({ error }: Props) => (
  <StyledAlert elevation={12} variant="filled" severity="error">
    <AlertTitle>Error</AlertTitle>
    {error}
  </StyledAlert>
);

export default ErrorMessage;
