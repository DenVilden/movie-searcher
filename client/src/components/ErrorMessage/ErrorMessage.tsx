/* eslint-disable @typescript-eslint/no-non-null-assertion */
import Error from "next/error";
import { ApolloError } from "@apollo/client";

type Props = {
  error: ApolloError;
};

const ErrorMessage = ({ error }: Props) => (
  <Error
    statusCode={
      error.graphQLErrors?.length
        ? error.graphQLErrors[0].extensions!.response.status
        : // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          error.networkError!.statusCode
    }
    title={
      error.graphQLErrors?.length
        ? error.graphQLErrors[0].extensions!.response.statusText
        : error.networkError?.message
    }
  />
);

export default ErrorMessage;
