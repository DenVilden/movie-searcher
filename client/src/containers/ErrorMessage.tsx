import React from 'react';
import styled from 'styled-components';
import { Typography, Theme } from '@material-ui/core';

type Props = {
  children: string;
  gutterBottom?: boolean;
};

const ErrorMessage = styled(
  ({ children, gutterBottom, ...otherProps }: Props) => (
    <Typography
      align="center"
      color="error"
      gutterBottom={gutterBottom}
      role="errormessage"
      variant="h6"
      {...otherProps}
    >
      {children}
    </Typography>
  )
)`
  &.MuiTypography-root {
    padding: ${({ theme }: { theme: Theme }) => theme.spacing(4)}px;
  }
`;

export default ErrorMessage;
