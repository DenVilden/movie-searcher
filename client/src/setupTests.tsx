import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from './pages/_app';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
