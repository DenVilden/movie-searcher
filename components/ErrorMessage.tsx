import Alert from '@material-ui/core/Alert';
import AlertTitle from '@material-ui/core/AlertTitle';
import styled from '@emotion/styled';

const StyledAlert = styled(Alert)`
  margin: ${({ theme }) => theme.spacing(2, 2, 0, 2)};
`;

interface Props {
  error: string;
}

export default function ErrorMessage({ error }: Props) {
  return (
    <StyledAlert elevation={12} severity="error" variant="filled">
      <AlertTitle>Error</AlertTitle>
      {error}
    </StyledAlert>
  );
}
