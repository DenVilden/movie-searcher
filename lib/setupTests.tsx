/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/preact'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import '@testing-library/jest-dom/extend-expect'

const theme = createMuiTheme()

type RenderApolloOptions = {
  addTypeName?: boolean
  mocks?: MockedResponse[]
}

export function renderApollo(
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) {
  return render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  )
}

export { screen, fireEvent } from '@testing-library/preact'
