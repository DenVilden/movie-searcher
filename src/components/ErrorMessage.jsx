import React, { Suspense } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Typography } from '@material-ui/core';
import Spinner from './Spinner';

const propTypes = {
  children: PropTypes.string.isRequired,
};

const ErrorMessage = styled(({ children, ...otherProps }) => (
  <Suspense fallback={<Spinner />}>
    <Typography align="center" color="error" variant="h6" {...otherProps}>
      {children}
    </Typography>
  </Suspense>
))`
  &.MuiTypography-root {
    padding: ${({ theme }) => theme.spacing(4)}px;
  }
`;

ErrorMessage.propTypes = propTypes;

export default ErrorMessage;
