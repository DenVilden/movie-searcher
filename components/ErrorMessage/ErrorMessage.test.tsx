import { renderApollo } from '../../lib/setupTests';
import ErrorMessage from './ErrorMessage';

describe('errorMessage', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderApollo(<ErrorMessage error="error" />);

    const element = asFragment();

    expect(element).toMatchSnapshot();
  });
});
