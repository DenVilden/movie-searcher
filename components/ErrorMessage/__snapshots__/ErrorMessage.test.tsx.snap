// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorMessage should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiAlert-root MuiAlert-filledError ErrorMessage__StyledAlert-sc-1jr6wcj-0 hDeQXw MuiPaper-elevation12"
    role="alert"
  >
    <div
      class="MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message"
    >
      <div
        class="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Error
      </div>
      error
    </div>
  </div>
</DocumentFragment>
`;
